### PR TITLE
Check if in masternode mode first and only then do the job (or not)

### PR DESCRIPTION
**Pull request by UdjinM6**
Shouldn't change the actual behaviour, mostly to streamline the logic/cleanup the code here.

### CODE REVIEW

File: src/privatesend-client.cpp [hunk 1]
```diff
@@ -328,24 +328,24 @@ void CPrivateSendClient::CheckPool()
 //
 void CPrivateSendClient::CheckTimeout()
 {
+    if(fMasternodeMode) return;
+
     CheckQueue();
 
-    if(!fEnablePrivateSend && !fMasternodeMode) return;
+    if(!fEnablePrivateSend) return;
```
Comment (nmarley) [inline src/privatesend-client.cpp]: If `CheckQueue` is intensive (or maybe even not), would it be better to move this above that call?

File: src/privatesend-server.cpp [hunk 2]
```diff
@@ -275,28 +275,22 @@ void CPrivateSendServer::SetNull()
...
-        }
+    // If entries are full, create finalized transaction
+    if (nState == POOL_STATE_ACCEPTING_ENTRIES && GetEntriesCount() >= CPrivateSend::GetMaxPoolTransactions()) {
+        LogPrint("privatesend", "CPrivateSendServer::CheckPool -- FINALIZE TRANSACTIONS\n");
+        CreateFinalTransaction(connman);
+        return;
     }
 
-    // reset if we're here for 10 seconds
-    if((nState == POOL_STATE_ERROR || nState == POOL_STATE_SUCCESS) && GetTimeMillis() - nTimeLastSuccessfulStep >= 10000) {
```
Comment (nmarley) [inline src/privatesend-server.cpp]: Comment in PR conversation says shouldn't change the actual behaviour, but I noticed the timeout/reset bit was removed here, is this intentional or did it get overlooked?

File: src/privatesend-client.cpp (hunk 1 above)
Comment (UdjinM6) [reply src/privatesend-client.cpp]: Not very intensive (Queue TTL is low and there are not so many of them) but anyway, that's intentional. User can stop PrivateSend at any moment and we still have to check the queue vector and drop expired ones.

File: src/privatesend-server.cpp (hunk 2 above)
Comment (UdjinM6) [reply src/privatesend-server.cpp]: The reason for this removal is that this block was a copy/paste when we spit one huge privatesend file into multiple classes/files trying to make as little changes as possible to not to break anything. If you look at `CPrivateSendServer::SetState()` you'll notice that these states (`POOL_STATE_ERROR` and `POOL_STATE_SUCCESS`) are illegal for masternodes. However, I shouldn't include this in this PR probably. Will revert this and submit in a new PR.

Comment (UdjinM6) [reply src/privatesend-server.cpp]: Fixed + see https://github.com/dashpay/dash/pull/2009 :)

File: src/privatesend-server.cpp [hunk 3]
```diff
@@ -275,22 +275,22 @@ void CPrivateSendServer::SetNull()
 //
 void CPrivateSendServer::CheckPool(CConnman& connman)
 {
-    if(fMasternodeMode) {
-        LogPrint("privatesend", "CPrivateSendServer::CheckPool -- entries count %lu\n", GetEntriesCount());
+    if (!fMasternodeMode) return;
```
Comment (codablock) [inline src/privatesend-server.cpp]: I'm a big fan of early bailout instead of (nested) conditionals. So, :+1: :)

**Pull request merged**